src/Module/Conversation/Channel.php

```diff
@@ -0,0 +1,398 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * See update_profile.php for documentation
+ */
+
+namespace Friendica\Module\Conversation;
+
+use Friendica\BaseModule;
+use Friendica\Content\BoundariesPager;
+use Friendica\Content\Conversation;
+use Friendica\Content\Feature;
+use Friendica\Content\Nav;
+use Friendica\Content\Text\HTML;
+use Friendica\Content\Widget;
+use Friendica\Content\Widget\TrendingTags;
+use Friendica\Core\Cache\Enum\Duration;
+use Friendica\Core\Renderer;
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Model\Contact;
+use Friendica\Model\Post;
+use Friendica\Model\User;
+use Friendica\Module\Security\Login;
+use Friendica\Network\HTTPException;
+
+class Channel extends BaseModule
+{
+	const WHATSHOT  = 'whatshot';
+	const FORYOU    = 'foryou';
+	const FOLLOWERS = 'followers';
+	const IMAGE     = 'image';
+	const VIDEO     = 'video';
+	const AUDIO     = 'audio';
+	/**
+	 * @}
+	 */
+
+	protected static $content;
+	protected static $accountTypeString;
+	protected static $accountType;
+	protected static $itemsPerPage;
+	protected static $min_id;
+	protected static $max_id;
+	protected static $item_id;
+
+	protected function content(array $request = []): string
+	{
+		if (!DI::userSession()->getLocalUserId()) {
+			return Login::form();
+		}
+
+		$this->parseRequest();
+
+		$t = Renderer::getMarkupTemplate("community.tpl");
+		$o = Renderer::replaceMacros($t, [
+			'$content' => '',
+			'$header'  => '',
+		]);
+
+		if (DI::pConfig()->get(DI::userSession()->getLocalUserId(), 'system', 'infinite_scroll')) {
+			$tpl = Renderer::getMarkupTemplate('infinite_scroll_head.tpl');
+			$o .= Renderer::replaceMacros($tpl, ['$reload_uri' => DI::args()->getQueryString()]);
+		}
+
+		if (empty($_GET['mode']) || ($_GET['mode'] != 'raw')) {
```

Comment:
Please use `$request` instead of `$_GET` superglobal.

Reply:
Done